plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php

```diff
@@ -31,17 +31,6 @@ public function __construct() {
 		add_filter( 'woocommerce_payment_gateways', array( __CLASS__, 'possibly_register_pre_install_wc_pay_promotion_gateway' ) );
 		add_filter( 'option_woocommerce_gateway_order', array( __CLASS__, 'set_gateway_top_of_list' ) );
 		add_filter( 'default_option_woocommerce_gateway_order', array( __CLASS__, 'set_gateway_top_of_list' ) );
-
-		$rtl = is_rtl() ? '.rtl' : '';
-
-		wp_enqueue_style(
-			'wc-admin-payment-method-promotions',
-			WCAdminAssets::get_url( "payment-method-promotions/style{$rtl}", 'css' ),
-			array( 'wp-components' ),
-			WCAdminAssets::get_file_version( 'css' )
-		);
-
-		WCAdminAssets::register_script( 'wp-admin-scripts', 'payment-method-promotions', true );
```

Comment:
Moved this to `plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php`. 

This shouldn't be called in the constructor as it will be called multiple times. Additionally, observing it in watch dev mode sometimes results in errors.

```
[21-Feb-2024 07:09:48 UTC] PHP Fatal error:  Uncaught Exception: Could not find asset registry for wp-admin-scripts in /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php:136
Stack trace:
#0 /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php(471): Automattic\WooCommerce\Internal\Admin\WCAdminAssets::get_script_asset_filename('wp-admin-script...', 'payment-method-...')
#1 /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php(44): Automattic\WooCommerce\Internal\Admin\WCAdminAssets::register_script('wp-admin-script...', 'payment-method-...', true)
#2 /Users/chihsuan/Projects/woocommerce/plugins/woocommerce/src/Admin/Features/Features.php(138): Automattic\WooCommerce\Internal\Admin\WCPayPromotion\Init->__construct()
```

Reply:
Good catch!